kartothek/io_components/metapartition.py

```diff
@@ -1499,7 +1499,9 @@ def _ensure_native_string_type(obj):
         return str(obj)
 
 
-def parse_input_to_metapartition(obj, metadata_version=None):
+def parse_input_to_metapartition(
+    obj, metadata_version=None, assert_partition_level_indices=False
```

Comment:
What's a `partition_level` index? We should probably stick to a nomenclature and we're usually referring to these as `secondary_indices`. Is there a reason why you changed this?

Reply:
I personally made a difference here between how indices are created, either using the`secondary_indices` kwarg or using what I termed "partition-level" indices: created using the `indices` key as in
```python
            {
                "label": "cluster_2",
                "data": [("core", df2)],
                "indices": {
                    "P": ExplicitSecondaryIndex(
                        "P", {v: ["cluster_2"] for v in values_p2}
                    )
                },
            },
```
Thus, I tried to emphasize that the the indices being checked in this commit are not those of the `secondary_indices` kwarg, but the indices of the kind defined in the code snippet. 
Anyways, I agree with the nomenclature point, so if you think this should be called `assert_secondary_indices` I will rename it